src/modules/ekf2/EKF2.cpp

```diff
@@ -2411,6 +2412,16 @@ void EKF2::UpdateGpsSample(ekf2_timestamps_s &ekf2_timestamps)
 			return; //TODO: change and set to NAN
 		}
 
+#if defined(CONFIG_EKF2_GNSS_YAW)
+
+		if (std::isnan(vehicle_gps_position.heading_offset) && PX4_ISFINITE(vehicle_gps_position.heading)) {
```

Comment:
I would structure it so that this is only done if `fabsf(_param_ekf2_gps_yaw_off.get()) > 0.f`, otherwise leave the fields alone.

Reply:
What if the param value is actual 0? The rover and base could be positioned so that the offset is actually 0 degrees in the param

Reply:
fixed